esmvaltool/diag_scripts/emergent_constraints/ecs_scatter.py

```diff
@@ -0,0 +1,813 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Diagnostic script to calculate various emergent constraints for ECS.
+
+Description
+-----------
+Calculate the X-axis of various emergent constraint for the Equilibrium
+Climate Sensitivity (ECS).
+
+Author
+------
+Manuel Schlund (DLR, Germany)
+
+Project
+-------
+CRESCENDO
+
+Configuration options in recipe
+-------------------------------
+diag : str
+  Emergent constraint to calculate (must be one of ``'brient_shal'``, ``'su'``,
+  ``'volodin'``, ``'zhai'``.
+metric : str, optional (default: 'regression_slope')
+    Metric to measure model error. Only relevant for Su et al. (2014)
+    constraint. Must be one of ``'regression_slope'``,
+    ``'correlation_coefficient'``.
+n_jobs : int, optional (default: 1)
+    Maximum number of jobs spawned by this class.
+output_attributes : dict, optional
+    Write additional attributes to netcdf files.
+pattern : str, optional
+    Pattern matched against ancestor file names.
+savefig_kwargs : dict, optional
+    Keyword arguments for :func:`matplotlib.pyplot.savefig`.
+seaborn_settings : dict, optional
+    Options for :func:`seaborn.set` (affects all plots), see
+    <https://seaborn.pydata.org/generated/seaborn.set.html>.
+
+"""
+
+import logging
+import os
+from copy import deepcopy
+from inspect import isfunction
+from pprint import pformat
+
+import dask.array as da
+import iris
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from joblib import Parallel, delayed
+from scipy.interpolate import interp1d
+from scipy.stats import linregress
+
+import esmvaltool.diag_scripts.emergent_constraints as ec
+import esmvaltool.diag_scripts.shared.iris_helpers as ih
+from esmvalcore.cmor._fixes.shared import (add_plev_from_altitude,
+                                           add_sigma_factory)
```

Comment:
Please only import things from ESMValCore that are part of the public API, documented here: https://docs.esmvaltool.org/projects/esmvalcore/en/latest/api/esmvalcore.html
Using undocumented and private (i.e. modules/functions starting with an underscore) is not a good idea, because these can be changed without notice in a minor revision.

There are two possible ways to solve this:
1) make the functions you need part of the public API
2) copy paste the required code here

Reply:
Opened ESMValGroup/ESMValCore#714 to add these functions to the public API.

Reply:
Note that I changed the module path a bit, the functions are now available in [`esmvalcore.cmor.fixes`.](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/api/esmvalcore.cmor.html#module-esmvalcore.cmor.fixes)